src/conn/conn_layered.c

```diff
@@ -1155,17 +1167,16 @@ __layered_drain_ingest_tables(WT_SESSION_IMPL *session)
 
     WT_RET(__wt_open_internal_session(conn, "disagg-drain", false, 0, 0, &internal_session));
 
-    /*
-     * The table count never shrinks, so this is safe. It probably needs the layered table lock.
-     *
-     * TODO: skip empty ingest tables.
-     */
+    __wt_spin_lock(session, &manager->layered_table_lock);
```

Comment:
I don't love holding this lock across maintenance work, but since the maintenance work is a no-op at the moment we can solve that when we get to it.